src/NHibernate/Criterion/GroupedProjection.cs

```diff
@@ -25,6 +25,9 @@ public virtual SqlString ToGroupSqlString(ICriteria criteria, ICriteriaQuery cri
 		{
 			if (projection is IPropertyProjection propertyProjection)
 				return new SqlString(string.Join(", ", criteriaQuery.GetColumns(criteria, propertyProjection.PropertyName)));
+			// This is because if grouped by an entity, multiple columns are in an projection which all have as aliases, which are not allowed.
+			if (projection is EntityProjection)
+				return SqlStringHelper.RemoveAllAsAliasesFromSql(renderedProjection);
```

Comment:
With such specific `EntityProjection` check I would implement `EntityProjection.GetColumns` and call it instead (similar to how it's done for `IPropertyProjection` above)

Reply:
> With such specific `EntityProjection` check I would implement `EntityProjection.GetColumns` and call it instead (similar to how it's done for `IPropertyProjection` above)

Do you mean with 'I' me or would you implement it? Aside from that, could you lead me in the direction where I can find similar code.

Reply:
> Do you mean with 'I' me or would you implement it? 

He meant 'If I were you' or 'it would be better to'. So it implied that it is better to make the change.